src/dash-table/utils/actions.js

```diff
@@ -53,23 +61,36 @@ export const clearSelection = {
     selected_cells: []
 };
 
-export function editColumnName(column, columns, headerRowIndex) {
+export function changeColumnHeader(column, columns, headerRowIndex, mergeDuplicateHeaders, newColumnName) {
+    let cloneColumn = Object.assign({}, column);
```

Comment:
Use `R.merge` or `R.mergeAll` for consistency. The nice advantage is that it treats all arguments as immutables -- so you do not have to add a target `{ }`.